packages/editor/src/components/preview-dropdown/index.js

```diff
@@ -110,13 +159,38 @@ export default function PreviewDropdown( { forceIsAutosaveable, disabled } ) {
 	 */
 	const onSelect = ( value ) => {
 		setDeviceType( value );
+		let newEditorMode = originalEditingMode.current;
 		if ( value === 'Desktop' ) {
 			speak( __( 'Desktop selected' ), 'assertive' );
 		} else if ( value === 'Tablet' ) {
 			speak( __( 'Tablet selected' ), 'assertive' );
-		} else {
+		} else if ( value === 'Mobile' ) {
 			speak( __( 'Mobile selected' ), 'assertive' );
+		} else if ( value === 'ZoomIn' ) {
+			speak( __( 'Zoom to 100 percent selected' ), 'assertive' );
+		} else {
+			speak( __( 'Zoom to 50 percent selected' ), 'assertive' );
+			newEditorMode = 'zoom-out';
 		}
+
+		if ( isZoomOutExperiment ) {
+			__unstableSetEditorMode( newEditorMode );
+		}
+	};
+
+	const getIcon = () => {
+		if (
+			isZoomOutExperiment &&
+			deviceType !== 'Mobile' &&
+			deviceType !== 'Tablet'
+		) {
+			if ( editorMode === 'zoom-out' ) {
+				return <>{ __( '50%' ) }</>;
+			}
+			return <>{ __( '100%' ) }</>;
+		}
+
+		return deviceIcons[ deviceType.toLowerCase() ];
```

Comment:
You're setting the `deviceType` to `ZoomIn` or `ZoomOut` above and the options are only available in the experiment, so the complicated function isn't necessary.

```suggestion
	const getIcon = () => {
		switch (deviceType) {
			case 'ZoomOut':
				return __( '50%' );
			case 'ZoomIn':
				return __( '100%' );
			default:
				return deviceIcons[ deviceType.toLowerCase() ];
		}
	}
```